### DISCUSSION TITLE
use of drive letters on windows (and under WSL)

### DISCUSSION THREAD

**Discussion by masinter** [Interlisp/medley#1187] [Yearnings]
... and when you had a colon in the name NOT as part of a device it confused the parser.

When you're running on a Windows system, host {dsk} should have a device -- "{dsk}D:/foo/bar.txt" would be the way I would expect to refer to a directory on the D: drive.

_Originally posted by @nbriggs in https://github.com/Interlisp/medley/issues/651#issuecomment-1021916658_

Under WSL we could map the drive letters or "device" for remote mounts

{DSK}C:/path1/path2/file  =>  /mnt/c/path1/path2/file

**Comment by nbriggs** [comment]
If we were using a WSL Unix approximation of a Windows file name, then sure, the mapping DEVICE: to /mnt/*DEVICE*/path1/path2/file would probably be the right thing, otherwise if the code were compiled for the _Windows_ interfaces which have a drive letter, then we should consider what to do with it.   I don't know if we're going to run in pure *Windows* mode, though.

**Comment by masinter** [comment]
Going to transfer this to a Discussion about windows without X
